### PR TITLE
catalog grid was added

### PR DESCRIPTION
**Pull request by Yuliia-Hrinchenko1**
- [DEMO LINK](https://Yuliia-Hrinchenko1.github.io/layout_catalog_grid/)

### CODE REVIEW

File: src/styles/_catalog.scss [hunk 1]
```diff
@@ -1,15 +1,31 @@
 .catalog {
-  display: flex;
-  flex-wrap: wrap;
-
-  max-width: 944px;
+  display: grid;
+  gap: $gap;
+  max-width: catalogWidth($cardWidth,
+  $gap, $cardsInRow);
```
Comment (olecsiy-gavrilyuk) [inline src/styles/_catalog.scss:5]: Are you shure about this function?

File: src/styles/_catalog.scss [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
+  }
+
   &__card {
     display: flex;
     justify-content: center;
     align-items: center;
-    width: 200px;
+    width: $cardWidth;
     height: 300px;
     border: 1px solid #ccc;
```
Comment (olecsiy-gavrilyuk) [inline src/styles/_catalog.scss:30]: you can add variable for this border color

**Review by olecsiy-gavrilyuk — CHANGES_REQUESTED** [review]
Good job)

File: src/styles/_catalog.scss (hunk 2 above)
Comment (Yuliia-Hrinchenko1) [reply src/styles/_catalog.scss:30]: I don't want to overuse variables) 
image

File: src/styles/blocks/catalog.scss [hunk 3]
```diff
@@ -0,0 +1,29 @@
+.catalog {
+  display: grid;
+  gap: $gap;
+  grid-template-columns: $cardWidth;
+  justify-content: center;
+  margin: 0 auto;
```
Comment (sergii-nosachenko) [inline src/styles/blocks/catalog.scss:6]:
```suggestion
```
1 - BEM
2 - this does nothing (you've already centered grid)

File: src/styles/utils/_variables.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+$gap: 48px;
+$cardWidth: 200px;
+$cardsInRow: 4;
```
Comment (sergii-nosachenko) [inline src/styles/utils/_variables.scss:3]: Unused variable

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]
Small fixes are needed.

File: src/styles/blocks/catalog.scss [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    grid-template-columns: repeat(2, $cardWidth);
+  }
+
+  @media (min-width: 768px) {
+    grid-template-columns: repeat(3, $cardWidth);
+  }
+
+  @media (min-width: 1024px) {
+    grid-template-columns: repeat(4, $cardWidth);
+  }
```
Comment (sergii-nosachenko) [inline src/styles/blocks/catalog.scss:17]: You can make a loop to create all media queries. Just create a correctly mapped list to iterate.

File: src/styles/blocks/catalog.scss [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+  &__card {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: $cardWidth;
```
Comment (sergii-nosachenko) [inline src/styles/blocks/catalog.scss:23]: You don't need to set the width for the card cause it is already set for the column.

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

**Pull request closed without merging**